TM4J használatakor MissingResourceException

TM4J-vel teszteket futtatva az alábbi runtime exception keletkezik (`Can't find bundle for base name javax.servlet.LocalStrings, locale en_US`):
```java
javax.enterprise.inject.CreationException
        at java.base/java.lang.J9VMInternals.newInstanceImpl(Native Method)
        at java.base/java.lang.Class.newInstance(Class.java:2086)
        at org.jboss.weld.security.NewInstanceAction.run(NewInstanceAction.java:33)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:734)
        at org.jboss.weld.injection.Exceptions.rethrowException(Exceptions.java:40)
        at org.jboss.weld.injection.Exceptions.rethrowException(Exceptions.java:78)
        at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:103)
        at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:85)
        at org.jboss.weld.injection.producer.ProducerMethodProducer.produce(ProducerMethodProducer.java:103)
        at org.jboss.weld.injection.producer.AbstractMemberProducer.produce(AbstractMemberProducer.java:161)
        at org.jboss.weld.bean.AbstractProducerBean.create(AbstractProducerBean.java:180)
        at org.jboss.weld.contexts.unbound.DependentContextImpl.get(DependentContextImpl.java:64)
        at org.jboss.weld.bean.ContextualInstanceStrategy$DefaultContextualInstanceStrategy.get(ContextualInstanceStrategy.java:100)
        at org.jboss.weld.bean.ContextualInstance.get(ContextualInstance.java:50)
        at org.jboss.weld.manager.BeanManagerImpl.getReference(BeanManagerImpl.java:694)
        at org.jboss.weld.bean.builtin.InstanceImpl.getBeanInstance(InstanceImpl.java:252)
        at org.jboss.weld.bean.builtin.InstanceImpl.get(InstanceImpl.java:114)
        at hu.icellmobilsoft.roaster.tm4j.junit5.Tm4jExtension.lambda$new$0(Tm4jExtension.java:62)
        at hu.icellmobilsoft.roaster.tm4j.junit5.Tm4jExtension$$Lambda$381/0000000000000000.get(Unknown Source)
        at hu.icellmobilsoft.roaster.tm4j.junit5.Tm4jExtension.getReporter(Tm4jExtension.java:102)
        at hu.icellmobilsoft.roaster.tm4j.junit5.Tm4jExtension.testSuccessful(Tm4jExtension.java:81)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$nodeFinished$15(TestMethodTestDescriptor.java:309)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor$$Lambda$612/0000000000000000.accept(Unknown Source)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestWatchers$17(TestMethodTestDescriptor.java:336)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor$$Lambda$613/0000000000000000.accept(Unknown Source)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestWatchers(TestMethodTestDescriptor.java:334)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.nodeFinished(TestMethodTestDescriptor.java:306)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.nodeFinished(TestMethodTestDescriptor.java:71)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.reportCompletion(NodeTestTask.java:172)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:85)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService$$Lambda$234/0000000000000000.accept(Unknown Source)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$230/0000000000000000.execute(Unknown Source)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$229/0000000000000000.invoke(Unknown Source)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$228/0000000000000000.execute(Unknown Source)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService$$Lambda$234/0000000000000000.accept(Unknown Source)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$230/0000000000000000.execute(Unknown Source)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$229/0000000000000000.invoke(Unknown Source)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$228/0000000000000000.execute(Unknown Source)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
        at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
        at org.junit.platform.launcher.core.DefaultLauncher$$Lambda$199/0000000000000000.accept(Unknown Source)
        at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:142)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:117)
        at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: java.lang.ExceptionInInitializerError
        at java.base/java.lang.J9VMInternals.ensureError(J9VMInternals.java:186)
        at java.base/java.lang.J9VMInternals.recordInitializationFailure(J9VMInternals.java:175)
        at java.base/java.lang.Class.forNameImpl(Native Method)
        at java.base/java.lang.Class.forName(Class.java:342)
        at org.jboss.resteasy.core.ContextParameterInjector$1.run(ContextParameterInjector.java:61)
        at org.jboss.resteasy.core.ContextParameterInjector$1.run(ContextParameterInjector.java:54)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:678)
        at org.jboss.resteasy.core.ContextParameterInjector.<clinit>(ContextParameterInjector.java:53)
        at org.jboss.resteasy.core.InjectorFactoryImpl.createParameterExtractor0(InjectorFactoryImpl.java:254)
        at org.jboss.resteasy.core.InjectorFactoryImpl.lambda$createParameterExtractor$1(InjectorFactoryImpl.java:158)
        at org.jboss.resteasy.core.InjectorFactoryImpl$$Lambda$501/000000002D935D30.apply(Unknown Source)
        at org.jboss.resteasy.core.InjectorFactoryImpl$OptionalInjections.lambda$wrap$8(InjectorFactoryImpl.java:309)
        at org.jboss.resteasy.core.InjectorFactoryImpl$OptionalInjections$$Lambda$516/000000002D915B70.get(Unknown Source)
        at java.base/java.util.Optional.orElseGet(Optional.java:369)
        at org.jboss.resteasy.core.InjectorFactoryImpl$OptionalInjections.wrap(InjectorFactoryImpl.java:309)
        at org.jboss.resteasy.core.InjectorFactoryImpl.createParameterExtractor(InjectorFactoryImpl.java:158)
        at org.jboss.resteasy.core.PropertyInjectorImpl.getParameterExtractor(PropertyInjectorImpl.java:133)
        at org.jboss.resteasy.core.PropertyInjectorImpl.populateMap(PropertyInjectorImpl.java:73)
        at org.jboss.resteasy.core.PropertyInjectorImpl.<init>(PropertyInjectorImpl.java:61)
        at org.jboss.resteasy.core.InjectorFactoryImpl.createPropertyInjector(InjectorFactoryImpl.java:73)
        at org.jboss.resteasy.core.providerfactory.Utils.injectProperties(Utils.java:75)
        at org.jboss.resteasy.core.providerfactory.CommonProviders.addMessageBodyReader(CommonProviders.java:193)
        at org.jboss.resteasy.core.providerfactory.CommonProviders.processProviderContracts(CommonProviders.java:87)
        at org.jboss.resteasy.core.providerfactory.ClientHelper.processProviderContracts(ClientHelper.java:104)
        at org.jboss.resteasy.core.providerfactory.ResteasyProviderFactoryImpl.processProviderContracts(ResteasyProviderFactoryImpl.java:841)
        at org.jboss.resteasy.core.providerfactory.ResteasyProviderFactoryImpl.registerProvider(ResteasyProviderFactoryImpl.java:829)
        at org.jboss.resteasy.core.providerfactory.ResteasyProviderFactoryImpl.registerProvider(ResteasyProviderFactoryImpl.java:816)
        at org.jboss.resteasy.plugins.providers.RegisterBuiltin.registerProviders(RegisterBuiltin.java:109)
        at org.jboss.resteasy.plugins.providers.RegisterBuiltin.register(RegisterBuiltin.java:74)
        at org.jboss.resteasy.plugins.providers.RegisterBuiltin.getClientInitializedResteasyProviderFactory(RegisterBuiltin.java:54)
        at org.jboss.resteasy.client.jaxrs.internal.ResteasyClientBuilderImpl.getProviderFactory(ResteasyClientBuilderImpl.java:377)
        at org.jboss.resteasy.microprofile.client.RestClientBuilderImpl.<init>(RestClientBuilderImpl.java:101)
        at org.jboss.resteasy.microprofile.client.BuilderResolver.newBuilder(BuilderResolver.java:9)
        at org.eclipse.microprofile.rest.client.RestClientBuilder.newBuilder(RestClientBuilder.java:55)
        at org.jboss.resteasy.microprofile.client.RestClientDelegateBean.create(RestClientDelegateBean.java:118)
        at org.jboss.weld.contexts.unbound.DependentContextImpl.get(DependentContextImpl.java:64)
        at org.jboss.weld.bean.ContextualInstanceStrategy$DefaultContextualInstanceStrategy.get(ContextualInstanceStrategy.java:100)
        at org.jboss.weld.bean.ContextualInstance.get(ContextualInstance.java:50)
        at org.jboss.weld.manager.BeanManagerImpl.getReference(BeanManagerImpl.java:694)
        at org.jboss.weld.manager.BeanManagerImpl.getInjectableReference(BeanManagerImpl.java:794)
        at org.jboss.weld.injection.FieldInjectionPoint.inject(FieldInjectionPoint.java:92)
        at org.jboss.weld.util.Beans.injectBoundFields(Beans.java:336)
        at org.jboss.weld.util.Beans.injectFieldsAndInitializers(Beans.java:347)
        at org.jboss.weld.injection.producer.ResourceInjector$1.proceed(ResourceInjector.java:69)
        at org.jboss.weld.injection.InjectionContextImpl.run(InjectionContextImpl.java:48)
        at org.jboss.weld.injection.producer.ResourceInjector.inject(ResourceInjector.java:71)
        at org.jboss.weld.injection.producer.BasicInjectionTarget.inject(BasicInjectionTarget.java:117)
        at org.jboss.weld.bean.ManagedBean.create(ManagedBean.java:161)
        at org.jboss.weld.contexts.unbound.DependentContextImpl.get(DependentContextImpl.java:64)
        at org.jboss.weld.bean.ContextualInstanceStrategy$DefaultContextualInstanceStrategy.get(ContextualInstanceStrategy.java:100)
        at org.jboss.weld.bean.ContextualInstance.get(ContextualInstance.java:50)
        at org.jboss.weld.manager.BeanManagerImpl.getReference(BeanManagerImpl.java:694)
        at org.jboss.weld.manager.BeanManagerImpl.getInjectableReference(BeanManagerImpl.java:794)
        at org.jboss.weld.injection.FieldInjectionPoint.inject(FieldInjectionPoint.java:92)
        at org.jboss.weld.util.Beans.injectBoundFields(Beans.java:336)
        at org.jboss.weld.util.Beans.injectFieldsAndInitializers(Beans.java:347)
        at org.jboss.weld.injection.producer.ResourceInjector$1.proceed(ResourceInjector.java:69)
        at org.jboss.weld.injection.InjectionContextImpl.run(InjectionContextImpl.java:48)
        at org.jboss.weld.injection.producer.ResourceInjector.inject(ResourceInjector.java:71)
        at org.jboss.weld.injection.producer.BasicInjectionTarget.inject(BasicInjectionTarget.java:117)
        at org.jboss.weld.bean.ManagedBean.create(ManagedBean.java:161)
        at org.jboss.weld.contexts.unbound.DependentContextImpl.get(DependentContextImpl.java:64)
        at org.jboss.weld.bean.ContextualInstanceStrategy$DefaultContextualInstanceStrategy.get(ContextualInstanceStrategy.java:100)
        at org.jboss.weld.bean.ContextualInstance.get(ContextualInstance.java:50)
        at org.jboss.weld.manager.BeanManagerImpl.getReference(BeanManagerImpl.java:694)
        at org.jboss.weld.bean.builtin.InstanceImpl.getBeanInstance(InstanceImpl.java:252)
        at org.jboss.weld.bean.builtin.InstanceImpl.get(InstanceImpl.java:114)
        at hu.icellmobilsoft.roaster.tm4j.common.Tm4jReporterProducer.createReporter(Tm4jReporterProducer.java:53)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:95)
        ... 67 more
Caused by: java.util.MissingResourceException: Can't find bundle for base name javax.servlet.LocalStrings, locale en_US
        at java.base/java.util.ResourceBundle.throwMissingResourceException(ResourceBundle.java:2055)
        at java.base/java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1689)
        at java.base/java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1593)
        at java.base/java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1556)
        at java.base/java.util.ResourceBundle.getBundle(ResourceBundle.java:857)
        at javax.servlet.ServletOutputStream.<clinit>(ServletOutputStream.java:88)
        ... 138 more
```

--- a/.github/workflows/compile_maven.yml
+++ b/.github/workflows/compile_maven.yml
@@ -25,4 +25,4 @@ jobs:
       with:
         java-version: 11
     - name: Build with Maven
-      run: mvn -B package --file pom.xml
+      run: mvn -B -Pci verify --file pom.xml

--- a/.github/workflows/compile_maven.yml
+++ b/.github/workflows/compile_maven.yml
@@ -25,4 +25,4 @@ jobs:
       with:
         java-version: 11
     - name: Build with Maven
-      run: mvn -B -Pci verify --file pom.xml
+      run: mvn -B verify --file pom.xml

--- a/docs/migration/migration060to070.adoc
+++ b/docs/migration/migration060to070.adoc
@@ -10,3 +10,10 @@
 === Átállás
 A változtatások nem eredményeznek átállási munkálatokat, visszafelé kompatibilis.
 
+
+=== roaster-tm4j
+* A `MissingResourceException` javításra került
+
+=== Átállás
+A változtatások nem eredményeznek átállási munkálatokat, visszafelé kompatibilis.
+

--- a/pom.xml
+++ b/pom.xml
@@ -29,6 +29,7 @@
 		<module>roaster-tm4j</module>
 		<module>roaster-hibernate</module>
 		<module>roaster-mongodb</module>
+		<module>roaster-testsuite</module>
 	</modules>
 
 	<properties>
@@ -279,13 +280,6 @@
 					</plugin>
 				</plugins>
 			</build>
-		</profile>
-
-		<profile>
-			<id>ci</id>
-			<modules>
-				<module>roaster-testsuite</module>
-			</modules>
 		</profile>
 	</profiles>
 </project>

--- a/pom.xml
+++ b/pom.xml
@@ -280,5 +280,12 @@
 				</plugins>
 			</build>
 		</profile>
+
+		<profile>
+			<id>ci</id>
+			<modules>
+				<module>roaster-testsuite</module>
+			</modules>
+		</profile>
 	</profiles>
 </project>

--- a/roaster-bom/pom.xml
+++ b/roaster-bom/pom.xml
@@ -14,6 +14,7 @@
 		<version.org.eclipse.microprofile>4.0</version.org.eclipse.microprofile>
 		<version.hu.icellmobilsoft.coffee>1.8.0</version.hu.icellmobilsoft.coffee>
 		<version.org.junit>5.6.2</version.org.junit>
+		<version.testcontainers>1.16.3</version.testcontainers>
 
 		<version.org.apache.commons.commons-text>1.8</version.org.apache.commons.commons-text>
 		<!-- rest-assured 4.0.0+ bug https://github.com/rest-assured/rest-assured/issues/1175 -->
@@ -195,6 +196,23 @@
 				<groupId>org.mockito</groupId>
 				<artifactId>mockito-core</artifactId>
 				<version>${version.org.mockito.mockito-core}</version>
+			</dependency>
+
+			<!-- testcontainers -->
+			<dependency>
+				<groupId>org.testcontainers</groupId>
+				<artifactId>testcontainers</artifactId>
+				<version>${version.testcontainers}</version>
+			</dependency>
+			<dependency>
+				<groupId>org.testcontainers</groupId>
+				<artifactId>junit-jupiter</artifactId>
+				<version>${version.testcontainers}</version>
+			</dependency>
+			<dependency>
+				<groupId>org.testcontainers</groupId>
+				<artifactId>mockserver</artifactId>
+				<version>${version.testcontainers}</version>
 			</dependency>
 
 			<dependency>

--- a/roaster-bom/pom.xml
+++ b/roaster-bom/pom.xml
@@ -15,6 +15,7 @@
 		<version.hu.icellmobilsoft.coffee>1.8.0</version.hu.icellmobilsoft.coffee>
 		<version.org.junit>5.6.2</version.org.junit>
 		<version.testcontainers>1.16.3</version.testcontainers>
+		<version.mockserver-client-java>5.11.2</version.mockserver-client-java>
 
 		<version.org.apache.commons.commons-text>1.8</version.org.apache.commons.commons-text>
 		<!-- rest-assured 4.0.0+ bug https://github.com/rest-assured/rest-assured/issues/1175 -->
@@ -30,6 +31,7 @@
 		<version.org.jboss.weld.weld-junit5>2.0.1.Final</version.org.jboss.weld.weld-junit5>
 		<version.org.jboss.weld.core>3.1.5.Final</version.org.jboss.weld.core>
 		<version.org.jboss.resteasy.client.microprofile>4.7.5.Final</version.org.jboss.resteasy.client.microprofile>
+		<version.javax.servlet>4.0.1</version.javax.servlet>
 		<version.commons.logging>1.2</version.commons.logging>
 
 		<!-- Selenide -->
@@ -198,23 +200,6 @@
 				<version>${version.org.mockito.mockito-core}</version>
 			</dependency>
 
-			<!-- testcontainers -->
-			<dependency>
-				<groupId>org.testcontainers</groupId>
-				<artifactId>testcontainers</artifactId>
-				<version>${version.testcontainers}</version>
-			</dependency>
-			<dependency>
-				<groupId>org.testcontainers</groupId>
-				<artifactId>junit-jupiter</artifactId>
-				<version>${version.testcontainers}</version>
-			</dependency>
-			<dependency>
-				<groupId>org.testcontainers</groupId>
-				<artifactId>mockserver</artifactId>
-				<version>${version.testcontainers}</version>
-			</dependency>
-
 			<dependency>
 				<groupId>com.fasterxml.jackson.datatype</groupId>
 				<artifactId>jackson-datatype-jsr310</artifactId>
@@ -247,10 +232,16 @@
 				<artifactId>weld-core-impl</artifactId>
 				<version>${version.org.jboss.weld.core}</version>
 			</dependency>
+
 			<dependency>
 				<groupId>org.jboss.resteasy</groupId>
 				<artifactId>resteasy-client-microprofile</artifactId>
 				<version>${version.org.jboss.resteasy.client.microprofile}</version>
+			</dependency>
+			<dependency>
+				<groupId>javax.servlet</groupId>
+				<artifactId>javax.servlet-api</artifactId>
+				<version>${version.javax.servlet}</version>
 			</dependency>
 
 			<!-- Oracle Driver -->
@@ -285,6 +276,31 @@
 				<version>${version.org.hibernate}</version>
 			</dependency>
 
+			<!-- testcontainers -->
+			<dependency>
+				<groupId>org.testcontainers</groupId>
+				<artifactId>testcontainers</artifactId>
+				<version>${version.testcontainers}</version>
+				<scope>test</scope>
+			</dependency>
+			<dependency>
+				<groupId>org.testcontainers</groupId>
+				<artifactId>junit-jupiter</artifactId>
+				<version>${version.testcontainers}</version>
+				<scope>test</scope>
+			</dependency>
+			<dependency>
+				<groupId>org.testcontainers</groupId>
+				<artifactId>mockserver</artifactId>
+				<version>${version.testcontainers}</version>
+				<scope>test</scope>
+			</dependency>
+			<dependency>
+				<groupId>org.mock-server</groupId>
+				<artifactId>mockserver-client-java</artifactId>
+				<version>${version.mockserver-client-java}</version>
+				<scope>test</scope>
+			</dependency>
 
 		</dependencies>
 	</dependencyManagement>

--- a/roaster-testsuite/pom.xml
+++ b/roaster-testsuite/pom.xml
@@ -1,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>roaster</artifactId>
+        <groupId>hu.icellmobilsoft.roaster</groupId>
+        <version>0.7.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>roaster-testsuite</artifactId>
+    <packaging>pom</packaging>
+
+    <modules>
+        <module>roaster-testsuite-tm4j</module>
+    </modules>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>hu.icellmobilsoft.roaster</groupId>
+                <artifactId>roaster-bom</artifactId>
+                <version>${project.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-failsafe-plugin</artifactId>
+                    <version>3.0.0-M5</version>
+                    <executions>
+                        <execution>
+                            <phase>integration-test</phase>
+                            <goals>
+                                <goal>integration-test</goal>
+                                <goal>verify</goal>
+                            </goals>
+                        </execution>
+                    </executions>
+                    <configuration>
+                        <includes>
+                            <include>**/IT*.java</include>
+                            <include>**/*IT.java</include>
+                            <include>**/Test*.java</include>
+                            <include>**/*Test.java</include>
+                            <include>**/*Tests.java</include>
+                            <include>**/*TestCase.java</include>
+                        </includes>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+</project>

--- a/roaster-testsuite/roaster-testsuite-tm4j/pom.xml
+++ b/roaster-testsuite/roaster-testsuite-tm4j/pom.xml
@@ -49,7 +49,6 @@
         <dependency>
             <groupId>org.mock-server</groupId>
             <artifactId>mockserver-client-java</artifactId>
-            <version>5.11.2</version>
             <exclusions>
                 <exclusion>
                     <groupId>javax.servlet</groupId>

--- a/roaster-testsuite/roaster-testsuite-tm4j/pom.xml
+++ b/roaster-testsuite/roaster-testsuite-tm4j/pom.xml
@@ -1,0 +1,62 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>roaster-testsuite</artifactId>
+        <groupId>hu.icellmobilsoft.roaster</groupId>
+        <version>0.7.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>roaster-testsuite-tm4j</artifactId>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-failsafe-plugin</artifactId>
+            </plugin>
+        </plugins>
+    </build>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>hu.icellmobilsoft.roaster</groupId>
+            <artifactId>roaster-weldunit</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>hu.icellmobilsoft.roaster</groupId>
+            <artifactId>roaster-tm4j-junit5</artifactId>
+        </dependency>
+
+        <!-- testcontainers, mockserver -->
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>testcontainers</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>junit-jupiter</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>mockserver</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.mock-server</groupId>
+            <artifactId>mockserver-client-java</artifactId>
+            <version>5.11.2</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>javax.servlet</groupId>
+                    <artifactId>javax.servlet-api</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+    </dependencies>
+</project>

--- a/roaster-testsuite/roaster-testsuite-tm4j/src/test/java/hu/icellmobilsoft/roaster/testsuite/tm4j/Tm4jIT.java
+++ b/roaster-testsuite/roaster-testsuite-tm4j/src/test/java/hu/icellmobilsoft/roaster/testsuite/tm4j/Tm4jIT.java
@@ -1,0 +1,98 @@
+/*-
+ * #%L
+ * Coffee
+ * %%
+ * Copyright (C) 2020 - 2022 i-Cell Mobilsoft Zrt.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.roaster.testsuite.tm4j;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockserver.client.MockServerClient;
+import org.mockserver.matchers.TimeToLive;
+import org.mockserver.matchers.Times;
+import org.mockserver.model.Header;
+import org.mockserver.model.Headers;
+import org.mockserver.model.HttpRequest;
+import org.mockserver.model.HttpResponse;
+import org.mockserver.model.JsonPathBody;
+import org.mockserver.model.MediaType;
+import org.testcontainers.containers.MockServerContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import hu.icellmobilsoft.roaster.tm4j.common.api.TestCaseId;
+import hu.icellmobilsoft.roaster.tm4j.junit5.Tm4jExtension;
+import hu.icellmobilsoft.roaster.weldunit.BaseWeldUnitType;
+
+/**
+ * Starts a mockserver with testcontainers to validate the TM4J rest calls
+ * 
+ * @author martin.nagy
+ * @since 0.7.0
+ */
+@Tag("integration")
+@ExtendWith(Tm4jExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class Tm4jIT extends BaseWeldUnitType {
+
+    private static final MockServerContainer MOCK_SERVER = new MockServerContainer(DockerImageName.parse("mockserver/mockserver"));
+    private static MockServerClient MOCK_SERVER_CLIENT;
+
+    @BeforeAll
+    static void beforeAll() {
+        MOCK_SERVER.start();
+        MOCK_SERVER_CLIENT = new MockServerClient(MOCK_SERVER.getHost(), MOCK_SERVER.getServerPort());
+
+        String mockServerUrl = "http://" + MOCK_SERVER.getHost() + ":" + MOCK_SERVER.getServerPort();
+        System.setProperty("roaster.tm4j.server/mp-rest/url", mockServerUrl);
+
+        Headers headers = new Headers(new Header("Accept", "application/json"), new Header("Authorization", "Basic ZXhhbXBsZS11c2VyOnNlY3JldA=="));
+
+        MOCK_SERVER_CLIENT.when(HttpRequest.request(), Times.unlimited(), TimeToLive.unlimited(), -1)
+                .respond(HttpResponse.response().withStatusCode(404).withBody("Not Found"));
+
+        MOCK_SERVER_CLIENT.when(HttpRequest.request().withPath("/rest/api/2/myself").withHeaders(headers))
+                .respond(HttpResponse.response().withContentType(MediaType.APPLICATION_JSON).withBody("{\"key\":\"test-user-1\"}"));
+
+        MOCK_SERVER_CLIENT.when(HttpRequest.request().withPath("/rest/atm/1.0/testrun/XXX-C123").withHeaders(headers))
+                .respond(HttpResponse.response().withStatusCode(200));
+
+        MOCK_SERVER_CLIENT.when(HttpRequest.request().withPath("/rest/atm/1.0/testcase/XXX-T1").withHeaders(headers))
+                .respond(HttpResponse.response().withStatusCode(200));
+
+        MOCK_SERVER_CLIENT.when(HttpRequest.request().withMethod("POST").withPath("/rest/atm/1.0/testrun/XXX-C123/testresults").withHeaders(headers))
+                .respond(HttpResponse.response().withStatusCode(200));
+    }
+
+    @AfterAll
+    static void afterAll() {
+        MOCK_SERVER_CLIENT.verify(HttpRequest.request().withMethod("POST").withPath("/rest/atm/1.0/testrun/XXX-C123/testresults")
+                .withBody(JsonPathBody.jsonPath("$[0][?(@.testCaseKey == 'XXX-T1')]")));
+        MOCK_SERVER.close();
+    }
+
+    @Test
+    @TestCaseId("XXX-T1")
+    void dummyTest() {
+        assertEquals(2, 1 + 1);
+    }
+}

--- a/roaster-testsuite/roaster-testsuite-tm4j/src/test/resources/META-INF/roaster-defaults.yml
+++ b/roaster-testsuite/roaster-testsuite-tm4j/src/test/resources/META-INF/roaster-defaults.yml
@@ -1,0 +1,6 @@
+roaster:
+  tm4j:
+    enabled: true
+    projectKey: XXX
+    testCycleKey: XXX-C123
+    server/basicAuthToken: ZXhhbXBsZS11c2VyOnNlY3JldA==

--- a/roaster-tm4j/roaster-tm4j-common/pom.xml
+++ b/roaster-tm4j/roaster-tm4j-common/pom.xml
@@ -41,7 +41,11 @@
             <groupId>com.fasterxml.jackson.datatype</groupId>
             <artifactId>jackson-datatype-jsr310</artifactId>
         </dependency>
-
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>javax.servlet-api</artifactId>
+            <version>4.0.1</version>
+        </dependency>
         <dependency>
             <groupId>org.jboss.resteasy</groupId>
             <artifactId>resteasy-client-microprofile</artifactId>

--- a/roaster-tm4j/roaster-tm4j-common/pom.xml
+++ b/roaster-tm4j/roaster-tm4j-common/pom.xml
@@ -44,7 +44,6 @@
         <dependency>
             <groupId>javax.servlet</groupId>
             <artifactId>javax.servlet-api</artifactId>
-            <version>4.0.1</version>
         </dependency>
         <dependency>
             <groupId>org.jboss.resteasy</groupId>

